content/sensu-go/6.8/catalog/catalog-reference.md

```diff
@@ -0,0 +1,1235 @@
+          op: add
+          value:
+            api_version: core/v2
+            type: Pipeline
+            name: "[[alerts_pipeline]]"
+        - path: /spec/pipelines/-
+          op: add
+          value:
+            api_version: core/v2
+            type: Pipeline
+            name: "[[incidents_pipeline]]"
+        - path: /spec/output_metric_thresholds/0/thresholds/0/max
+          op: replace
+          value: "[[nginx_active_warn]]"
+        - path: /spec/output_metric_thresholds/1/thresholds/0/max
+          op: replace
+          value: "[[nginx_waiting_warn]]"
+  post_install:
+    - type: section
+      title: Success
+    - type: markdown
+      body: |
+        You enabled the NGINX Monitoring integration.
+        The `nginx-metrics` check will run for all Sensu agents with these subscriptions: [[subscriptions]].
+{{< /code >}}
+
+Sensu Catalog integration definitions resemble other Sensu resources, but Sensu Go does not process them directly.
+Instead, the [Sensu Catalog API][2] uses integration definitions along with the other files in the catalog repository, like READMEs and dashboard images, to generate a static API.
+
+## Private catalogs
+
+The [Sensu Catalog API][2] renders static HTTP API content that the Sensu web UI can consume.
+This means you can create a private enterprise catalog of custom integrations and make it available to users in the Sensu web UI.
+
+Consider forking the official Sensu Catalog repository, https://github.com/sensu/catalog, as a starting point for building your own private catalog.
+
+The Catalog API defines integrations globally rather than by namespace.
+When you create a private catalog, all integrations in your repository are available for all users across namespaces in the web UI.
+
+Read [Build a private catalog of Sensu integrations][17] for more information.
+
+## catalog-api command line interface tool
+
+{{% notice note %}}
+**NOTE**: The catalog-api tool is an alpha feature and may include breaking changes.
+{{% /notice %}}
+
+Sensu's [catalog-api][16] command line interface (CLI) tool uses the [Sensu Catalog API][2] to convert integration files into static API content that you can host on any HTTP web service.
+
+Use the catalog-api tool to [generate a local catalog API][18] for testing as you develop new integrations and to build and run a private catalog.
+Integration files must be stored in a repository that follows the required [organizational framework][16].
+
+The catalog-api tool is written in Go.
+
+### catalog-api subcommands
+
+The catalog-api tool provides the following subcommands:
+
+{{< code text >}}
+catalog-api catalog --help
+USAGE
+  catalog-api catalog [flags] <subcommand> [flags]
+
+SUBCOMMANDS
+  generate  Generate a static catalog API
+  validate  Validate a catalog directory and its integrations
+  server    Serves static catalog API for development purposes
+  preview   Serves static catalog API & preview catalog web application for development purposes
+
```

Comment:
What is the difference between the server and preview subcommands?

Reply:
The server subcommand simply starts a webserver to serve the catalog api (just a bunch of JSON files for the most part). In order to test the catalog while running the server subcommand, a user needs to set up a Sensu Backend and create a CatalogConfig resource to point to the webserver.

The preview subcommand does the same thing but also serves a preview dashboard that can communicate with Sensu Backend for easy development & testing of integrations (screenshot below). In this scenario, a user doesn't need to create a CatalogConfig resource or interact with the Sensu Backend's Web UI.

<img width="1212" alt="Screen Shot 2022-10-17 at 11 27 08" src="https://user-images.githubusercontent.com/120659/196255335-1dbe7e0d-7c8c-40b0-8318-a1404b421407.png">